src/detect.rs

```diff
@@ -18,3 +23,196 @@ pub enum DetectOutcome {
     Pass(BuildPlan),
     Fail,
 }
+
+enum BuildpackTomlArg<BM> {
+    Path(PathBuf),
+    Str(String),
+    BuildpackToml(BuildpackToml<BM>),
+}
+
+enum PlatformArg<P> {
+    Plat(P),
+    Dir(PathBuf),
```

Comment:
Naming & Consistency: In `BuildpackTomlArg`, we use `Path(PathBuf)` and here we use `Dir(PathBuf)`. One points to a file and one to a directory, but I think there is an opportunity to clarify naming here.

What about `File(PathBuf)` and `Directory(PathBuf)` respectively? Another alternative could be using `Path` for both.

Reply:
I like being more specific. I try to differentiate in ruby via naming conventions https://github.com/heroku/buildpacks-ruby#naming-conventions.

Reply:
I went with `File` and `Directory`